packages/web3-react/src/components/AlephiumConnect.tsx

```diff
@@ -163,23 +187,23 @@ export const AlephiumConnectProvider: React.FC<{ children?: React.ReactNode }> =
       }
       subscribeToTxStatus(options, txId, undefined, undefined, expectedConfirmations)
     },
-    [updateBalance, account?.network]
+    [updateBalance, connectContext.account?.network]
   )
 
   useEffect(() => {
-    updateBalance()
-  }, [updateBalance])
+    if (balance === undefined) {
+      updateBalance()
+    } else if (connectContext.account === undefined) {
+      setBalance(undefined)
+    }
+  }, [updateBalance, balance, connectContext.account])
```

Comment:
Although this won't lead to rerenders, the code here is not very good. The else branch is used to ensure that the balance is set to undefined when the wallet disconnects, or we could use a separate `useEffect`. WDYT?